File: backend/src/Data/Entities/Barberia.cs
namespace CitaCorte.Data.Entities;

public class Barberia
{
    public int Id { get; set; }
    public int UsuarioId { get; set; }
    public Usuario Usuario { get; set; } = null!;
    
    public string? NombreComercial { get; set; }
    public string? Descripcion { get; set; }
    public string? Direccion { get; set; }
    public string? Ciudad { get; set; }
    public string? Pais { get; set; }
    public string? CodigoPostal { get; set; }
    public decimal? Latitud { get; set; }
    public decimal? Longitud { get; set; }
    public string? TelefonoContacto { get; set; }
    public string? FotoPerfilUrl { get; set; }
    public string? FotoPortadaUrl { get; set; }
    public decimal? CalificacionPromedio { get; set; }
    public int TotalReservasCompletadas { get; set; } = 0;
    
    // Suscripción actual activa
    public int? SuscripcionActivaId { get; set; }
    public Suscripcion? SuscripcionActiva { get; set; }
    
    // Estado de aprobación de suscripción (para saber si puede afiliar)
    public bool SuscripcionAprobada { get; set; } = false;
    
    public DateTime FechaRegistro { get; set; } = DateTime.UtcNow;
    public bool Activo { get; set; } = true;

    // Navegación
    public ICollection<Servicio> Servicios { get; set; } = new List<Servicio>();
    public ICollection<Barbero> BarberosAfiliados { get; set; } = new List<Barbero>();
    public ICollection<SolicitudAfiliacion> SolicitudesRecibidas { get; set; } = new List<SolicitudAfiliacion>();
    public ICollection<Reserva> ReservasComoBarberia { get; set; } = new List<Reserva>();
    public ICollection<EstadisticaBarberia> Estadisticas { get; set; } = new List<EstadisticaBarberia>();
}

File: backend/Data/Models/Barberia.js
import mongoose from 'mongoose';

const barberiaSchema = new mongoose.Schema({
  usuario: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Usuario',
    required: true
  },
  nombreComercial: {
    type: String,
    required: [true, 'El nombre comercial es requerido'],
    trim: true,
    maxlength: [100, 'El nombre comercial no puede exceder 100 caracteres']
  },
  direccion: {
    calle: String,
    numero: String,
    colonia: String,
    ciudad: String,
    estado: String,
    codigoPostal: String,
    pais: {
      type: String,
      default: 'México'
    },
    coordenadas: {
      latitud: Number,
      longitud: Number
    }
  },
  descripcion: {
    type: String,
    trim: true,
    maxlength: [1000, 'La descripción no puede exceder 1000 caracteres']
  },
  fotos: [{
    url: String,
    descripcion: String
  }],
  telefono: {
    type: String,
    trim: true,
    maxlength: [20, 'El teléfono no puede exceder 20 caracteres']
  },
  horarios: [{
    diaSemana: {
      type: String,
      enum: ['lunes', 'martes', 'miercoles', 'jueves', 'viernes', 'sabado', 'domingo']
    },
    apertura: String, // Formato HH:mm
    cierre: String,   // Formato HH:mm
    activo: {
      type: Boolean,
      default: true
    }
  }],
  planSuscripcionActual: {
    plan: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'PlanSuscripcion'
    },
    fechaInicio: Date,
    fechaVencimiento: Date,
    estado: {
      type: String,
      enum: ['activo', 'vencido', 'pendiente_aprobacion'],
      default: 'pendiente_aprobacion' // Requiere aprobación
    }
  },
  barberosAfiliados: [{
    barbero: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'Barbero'
    },
    fechaAfiliacion: Date,
    activo: {
      type: Boolean,
      default: true
    }
  }],
  serviciosOfrecidos: [{
    tipo: mongoose.Schema.Types.ObjectId,
    ref: 'Servicio'
  }],
  calificacionPromedio: {
    type: Number,
    default: 0,
    min: [0, 'La calificación no puede ser menor a 0'],
    max: [5, 'La calificación no puede ser mayor a 5']
  },
  redesSociales: {
    facebook: String,
    instagram: String,
    twitter: String,
    whatsapp: String
  }
}, {
  timestamps: true
});

// Índice para búsquedas
barberiaSchema.index({ usuario: 1 });
barberiaSchema.index({ 'direccion.ciudad': 1 });
barberiaSchema.index({ nombreComercial: 'text' });
barberiaSchema.index({ descripcion: 'text' });

// Método para verificar si puede afiliar más barberos
barberiaSchema.methods.puedeAfiliarBarbero = function() {
  if (this.planSuscripcionActual?.estado !== 'activo') return false;
  
  const limite = this.planSuscripcionActual?.plan?.limiteBarberos || 0;
  const actuales = this.barberosAfiliados.filter(b => b.activo).length;
  
  return actuales < limite;
};

// Método para obtener número de barberos afiliados activos
barberiaSchema.virtual('numeroBarberosActivos').get(function() {
  return this.barberosAfiliados.filter(b => b.activo).length;
});

export default mongoose.model('Barberia', barberiaSchema);
